Modules/ITS/include/ITS/ITSDecodingErrorTask.h

```diff
@@ -58,12 +59,16 @@ class ITSDecodingErrorTask final : public TaskInterface
   static constexpr int NLayer = 7;
   static constexpr int NLayerIB = 3;
   const int StaveBoundary[NLayer + 1] = { 0, 12, 28, 48, 72, 102, 144, 192 };
+  const int ChipBoundary[NLayer + 1] = { 0, 108, 252, 432, 3120, 6480, 14712, 24120 };
   static constexpr int NFees = 48 * 3 + 144 * 2;
 
-  TH1D* mLinkErrorPlots;
+  int*** mChipErrorCount;
```

Comment:
I see that you store in this array the number of ChipErrors. 

Do you think that it is possible to keep this number directly in histogram? 

For example, you have the code:

mChipErrorVsChipid[ilayer]->SetBinContent(ichip + 1, ierror + 1, mChipErrorCount[ilayer][ichip][ierror])

Could you did it similar to:
 mChipErrorVsChipid[ilayer]->Fill(ichip + 1, ierror + 1)? 
 
 In such way you don't need to have an additional array of ChipErrors, and also you will be able to avoid aloop at lines 185-192

Reply:
ok, I got it!